### PR TITLE
Address no colon case

### PR DESCRIPTION
**Pull request by zhouzhuojie**
Address the no-colon basic auth case.

```
key:         => key
key          => key
```

### CODE REVIEW

**Review by rstratton — APPROVED** [review]

File: kong/plugins/basicauth2keyauth/handler.lua [hunk 1]
```diff
@@ -28,7 +28,7 @@ local function retrieve_credentials(request, header_name, conf)
     if m and m[1] then
       local decoded_basic = ngx.decode_base64(m[1])
       if decoded_basic then
-        local basic_parts, err = ngx.re.match(decoded_basic, "([^:]+):(.*)", "oj")
+        local basic_parts, err = ngx.re.match(decoded_basic, "([^\t\r\n:]+):?(.*)", "oj")
```
Comment (ziru) [inline kong/plugins/basicauth2keyauth/handler.lua]: i would vote for the original fix:
```
+        local basic_parts, err = ngx.re.match(decoded_basic, "([^:]+):?(.*)", "oj")
```
instead, fix the wrong base64 encoded value used in unit test.

```
>>> base64.b64decode('YXBpMTIzNDpwYXNzd29yZAo=')
b'api1234:password\n'
>>> base64.b64decode('YXBpMTIzNAo=')
b'api1234\n'
```
i guess when you generate the values in shell, you use `echo <value> | base64`.  that would encode a string with newline char at the end.

instead, you should use `echo -n <value> | base64`.

under python REPL:
```
>>> base64.b64encode(b'api1234:password')
b'YXBpMTIzNDpwYXNzd29yZA=='
>>> base64.b64encode(b'api1234')
b'YXBpMTIzNA=='
```

under shell:
```
➜  ~ echo -n api1234:password | base64
YXBpMTIzNDpwYXNzd29yZA==
➜  ~ echo -n api1234 | base64
YXBpMTIzNA==
```

File: spec/basicauth2keyauth/01-access_spec.lua [hunk 2]
```diff
@@ -62,6 +62,18 @@ describe("Plugin: key-auth (access)", function()
       local header_value = assert.request(res).has.header("apikey_from_basicauth")
       assert.equal("api1234", header_value)
     end)
+    it("generates nothing if basic auth has the password part", function()
+      local res = assert(client:send {
+        method = "GET",
+        path = "/request",
+        headers = {
+          ["Host"] = "host2.com",
+          ["Authorization"] = "Basic YXBpMTIzNDpwYXNzd29yZAo=" -- "api1234:password"
```
Comment (ziru) [inline spec/basicauth2keyauth/01-access_spec.lua]: => "Basic YXBpMTIzNDpwYXNzd29yZA=="

File: spec/basicauth2keyauth/01-access_spec.lua [hunk 3]
```diff
@@ -75,17 +87,18 @@ describe("Plugin: key-auth (access)", function()
...
     end)
-    it("generates nothing if basic auth has the password part", function()
+    it("generates new api_key header with sha256 value if basic auth is in the headers without colon", function()
       local res = assert(client:send {
         method = "GET",
         path = "/request",
         headers = {
           ["Host"] = "host2.com",
-          ["Authorization"] = "Basic YXBpMTIzNDpwYXNzd29yZAo=" -- "api1234:password"
+          ["Authorization"] = "Basic YXBpMTIzNAo=" -- "api1234"
```
Comment (ziru) [inline spec/basicauth2keyauth/01-access_spec.lua]: => "Basic YXBpMTIzNA=="

**Comment by zhouzhuojie** [comment]
@ziru Thanks, the `echo -n` is very helpful!

**Review by ziru — APPROVED** [review]

**Pull request merged**